### PR TITLE
Cleanup building

### PR DESCRIPTION
**Pull request by mfine**
Cleanup the building of buildroot. This still doesn't compose well, especially in the docker container - changing anything is going to invalidate the cache layer and rebuild everything. Which has been a significant barrier to development. But additionally, clean things up:

+ dockerignore .git directory
+ bring in buildroot as a submodule instead of cloning it in - you can bring in submodules with a shallow clone.
+ pickup a newer docker base image, and switch to debian.
+ get rid of `build.sh`
+ simplify dockerfile building without `build.sh` - set BR2_EXTERNAL env, filter output for travis only in docker builds (but let some output out to give an idea of some kind of progress)
+ use a simple Makefile instead of `build.sh`

/cc @fnoble @jacobmcnamee @denniszollo

### CODE REVIEW

**Comment by jacobmcnamee** [comment]
`configs/piksiv3_defconfig` is autogenerated and should not be modified

File: Dockerfile [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
   bc \
-  libssl-dev
+  cpio
 
 COPY . /app
 
-RUN /app/build.sh
-
+RUN make -C buildroot piksiv3_defconfig
+RUN HW_CONFIG=prod make -C buildroot 2>&1 | tee -a build.out | grep --line-buffered '^make'
```
Comment (jacobmcnamee) [inline Dockerfile]: Do you still need `-C buildroot` here if it is also in the Makefile?

Comment (mfine) [inline Dockerfile]: Good point - can just use the targets.

**Comment by jacobmcnamee** [comment]
Looks fine. I'm fine with making buildroot a submodule, but buildroot does then download required packages anyway. You may want to add some generated buildroot subdirectories to `.gitignore` (e.g. `build`)

**Comment by anonymous-travis-swiftnav** [comment]
## v1.0.0-branch-57-g00bb935
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-57-g00bb935/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.0.0-branch-57-g00bb935/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-57-g00bb935

**Comment by mfine** [comment]
Reverted `configs/piksiv3_defconfig`. Currently building in docker, so not seeing the generated directories to ignore - but happy to add them. Will build on linux to check them out.

**Comment by anonymous-travis-swiftnav** [comment]
## v1.0.0-branch-58-g94579f5
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-58-g94579f5/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.0.0-branch-58-g94579f5/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-58-g94579f5

**Comment by anonymous-travis-swiftnav** [comment]
## v1.0.0-branch-59-g0ef7312
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-59-g0ef7312/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.0.0-branch-59-g0ef7312/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-59-g0ef7312

**Pull request merged**